File: arrays/169. Majority Element/majorityElement.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counts = {}
        half = len(nums) // 2
        
        for num in nums:
            if num in counts:
                counts[num] += 1
            else:
                counts[num] = 1
        
        for k,v in counts.items():
            if v > half:
                return k

File: arrays/169. Majority Element/majorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let counts = {};
    let half = nums.length / 2;
    
    for (let num of nums) {
        if (num in counts) {
            counts[num] += 1;
        } else {
            counts[num] = 1;
        }
    }
    
    for (let [k, v] of Object.entries(counts)) {
        if (v > half) {
            return Number(k);
        }
    }
};